core/src/main/java/nl/nn/adapterframework/soap/WsdlGenerator.java

```diff
@@ -575,51 +575,40 @@ protected void types(XMLStreamWriter w) throws XMLStreamException, IOException,
 
 	/**
 	 * Outputs the 'messages' section.
-	 * @param w
-	 * @throws XMLStreamException
-	 * @throws IOException
-	 * @throws ConfigurationException
+	 * @param w Writer to which XML is written.
+	 * @throws XMLStreamException Thrown is there is an exception writing to stream.
+	 *
 	 */
-	protected void messages(XMLStreamWriter w) throws XMLStreamException, IOException, ConfigurationException {
-		List<QName> parts = new ArrayList<QName>();
-		if (inputHeaderElement != null && !inputHeaderIsOptional) {
-			parts.add(inputHeaderElement);
+	protected void messages(XMLStreamWriter w) throws XMLStreamException {
+		List<QName> parts = new ArrayList<>();
+		addHeaderElement(w, parts, inputHeaderElement, inputHeaderIsOptional, inputBodyElement, inputRoot);
+		if (outputValidator != null) {
+			parts.clear();
+			addHeaderElement(w, parts, outputHeaderElement, outputHeaderIsOptional, outputBodyElement, outputRoot);
 		}
-		if (inputBodyElement != null) {
-			parts.add(inputBodyElement);
+	}
+
+	private void addHeaderElement(XMLStreamWriter w, List<QName> parts, QName headerElement, boolean headerIsOptional, QName bodyElement, String root) throws XMLStreamException {
+		if (headerElement != null && !headerIsOptional) {
+			parts.add(headerElement);
 		}
-		message(w, inputRoot, parts);
-		if (inputHeaderIsOptional) {
-			parts.clear();
-			if (inputHeaderElement != null) {
-				parts.add(inputHeaderElement);
-			}
-			message(w, inputRoot + "_" + inputHeaderElement.getLocalPart(), parts);
+		if (bodyElement != null) {
+			parts.add(bodyElement);
 		}
-		if (outputValidator != null) {
+		message(w, root, parts);
+		if (headerIsOptional) {
 			parts.clear();
-			if (outputHeaderElement != null && !outputHeaderIsOptional) {
-				parts.add(outputHeaderElement);
-			}
-			if (outputBodyElement != null) {
-				parts.add(outputBodyElement);
-			}
-			message(w, outputRoot, parts);
-			if (outputHeaderIsOptional) {
-				parts.clear();
-				if (outputHeaderElement != null) {
-					parts.add(outputHeaderElement);
-				}
-				message(w, outputRoot + "_"+ outputHeaderElement.getLocalPart(), parts);
+			if (headerElement != null) {
+				parts.add(headerElement);
+				message(w, root + "_" + headerElement.getLocalPart(), parts);
 			}
 		}
 	}
 
 	protected void message(XMLStreamWriter w, String root, Collection<QName> parts) throws XMLStreamException {
 		if (!parts.isEmpty()) {
 			w.writeStartElement(WSDL_NAMESPACE, "message");
-			w.writeAttribute("name", "Message_" + root);
-			{
+			w.writeAttribute("name", "Message_" + root); {
```

Comment:
Waarom is dit block scoped?
Misschien is het handiger de accolade niet achter een andere methode te zetten.